pkg/clients/aws/route53.go

```diff
@@ -611,16 +611,29 @@ func getSTSCredentials(reqLogger logr.Logger, client *sts.STS, roleArn string, e
 		if err == nil {
 			break
 		}
-		if i == (assumeRolePollingRetries - 1) {
-			return nil, fmt.Errorf("timed out while assuming role %s", roleArn)
+		// Error was returned
+		if i >= (assumeRolePollingRetries - 1) {
+			// If we're out of retries, return the wrapped error
+			return nil, fmt.Errorf("failed to assumeRole, no retries left: %w", err)
```

Comment:
Suggestion to help our future selves - log an info message here that we're out of retries and are bailing on this function.

Info because the error is being passed back up which _should_ be handled appropriately by the caller function.